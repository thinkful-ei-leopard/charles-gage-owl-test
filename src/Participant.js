import React from 'react';
import './Participant.css';

export default function Participant(props) {
    let status;
    if (props.onStage === true) {
        status = 'on stage';
    } else if (props.onStage === false && props.inSession === true) {
        status = 'in session';
    } else {
        status = 'left session';
    }
  return (
    <div className="session-participant">
        <img src={props.avatar} alt={props.name} className="user-avatar" />
        <div className="participant-info">
            <h6 className="participant-name">{props.name}</h6>
            <div className="participant-status participant-status__green">
               {status}
            </div>
        </div>
    </div>
  );
}