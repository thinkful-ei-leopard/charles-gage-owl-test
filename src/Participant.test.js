import React from 'react';
import Participant from './Participant';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  // first create a DOM element to render the component into
  const div = document.createElement('div');
  ReactDOM.render(<Participant props={[{
        id: 1,
        name: 'Koren Templeton',
        avatar:
            'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
        inSession: true,
        onStage: true
    }]}/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(<Participant props={[{
        id: 1,
        name: 'Koren Templeton',
        avatar:
            'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
        inSession: true,
        onStage: true
    }]}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();  
  });
