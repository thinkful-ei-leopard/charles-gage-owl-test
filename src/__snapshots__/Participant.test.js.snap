// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="session-participant"
>
  <img
    className="user-avatar"
  />
  <div
    className="participant-info"
  >
    <h6
      className="participant-name"
    />
    <div
      className="participant-status participant-status__green"
    >
      left session
    </div>
  </div>
</div>
`;
